signal_id/mask.py

```diff
@@ -320,10 +321,93 @@ def close(self, struct=None, iterations=1):
         self._mask = nd.binary_closing(self._mask, structure=struct,
                                        iterations=iterations)
 
+    def remove_small_regions(self, area_threshold=None, beam=None):
+        '''
+        Remove 2D regions (per channel) based on their area. By default, this
+        removed regions smaller than the beam area.
+
+        Parameters
+        ----------
+        area_threshold : float, optional
+            Minimum pixel area to keep. Overrides beam argument.
+        beam : radio_beam.Beam, optional
+            Provide a Beam object to define the area threshold. By default,
+            a Beam object will be created from the information in the cube
+            WCS. Specifying a Beam object will override using the default
+            Beam object from the cube WCS.
+
+        '''
+
+        self.log_and_backup(self.remove_small_regions)
+
+        # Attempt to get beam area from cube WCS info.
+        if area_threshold is None:
+            if beam is None:
+                beam = Beam.from_fits_header(self.cube.wcs.to_header())
```

Comment:
shouldn't the `cube` have a header?  I don't think you need to recreate one from `wcs`